tests/Unit/Evolution/Systems/Cce/Test_Equations.cpp

```diff
@@ -0,0 +1,199 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <complex>
+#include <cstddef>
+#include <string>
+#include <utility>
+
+#include "DataStructures/ComplexDataVector.hpp"
+#include "DataStructures/Variables.hpp"  // IWYU pragma: keep
+#include "Evolution/Systems/Cce/Equations.hpp"
+#include "Evolution/Systems/Cce/TagLists.hpp"
+#include "Evolution/Systems/Cce/Tags.hpp"  // IWYU pragma: keep
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "tests/Unit/Evolution/Systems/Cce/TestHelpers.hpp"  // IWYU pragma: keep
+#include "tests/Unit/Pypp/CheckWithRandomValues.hpp"
+#include "tests/Unit/Pypp/SetupLocalPythonEnvironment.hpp"
+
+// IWYU pragma: no_forward_declare Cce::Tags::Beta
+// IWYU pragma: no_forward_declare Cce::Tags::H
+// IWYU pragma: no_forward_declare Cce::Tags::Q
+// IWYU pragma: no_forward_declare Cce::Tags::U
+// IWYU pragma: no_forward_declare Cce::Tags::W
+// IWYU pragma: no_forward_declare Cce::Tags::Integrand
+// IWYU pragma: no_forward_declare Cce::Tags::LinearFactor
+// IWYU pragma: no_forward_declare Cce::Tags::LinearFactorForConjugate
+// IWYU pragma: no_forward_declare Cce::Tags::PoleOfIntegrand
+// IWYU pragma: no_forward_declare Cce::Tags::RegularIntegrand
+// IWYU pragma: no_forward_declare Cce::TagsCategory::IndependentOfBondiIntegration
+// IWYU pragma: no_forward_declare Cce::TagsCategory::SwshDerivatives
+// IWYU pragma: no_forward_declare Cce::TagsCategory::TagsToSwshDifferentiate
+// IWYU pragma: no_forward_declare Cce::TagsCategory::Temporary
+// IWYU pragma: no_forward_declare Cce::ComputeBondiIntegrand
+// IWYU pragma: no_forward_declare Variables
+
+namespace Cce {
+
+namespace {
+
+template <typename Type, size_t N>
+using ForwardIgnoreIndex = Type;
+
+template <typename Tag, typename VariablesTagList, typename Arg>
+int assign_to_tag(
+    const gsl::not_null<Variables<VariablesTagList>*> all_variables,
+    const Arg arg) noexcept {
+  get(get<Tag>(*all_variables)).data() = arg;
```

Comment:
Why do you need the `data()`? Looks dangerous :D

Reply:
This is a result of the SpinWeighted wrapper. The outer `get` obtains a SpinWeighted<ComplexDataVector, N>`, and the `.data()` obtains the `ComplexDataVector`, which can be assigned regardless of the spin-weight. It's not a pointer, which I suspect is why it feels concerning.